mantidproject/mantid
# Total Scattering test tidy up

**Description of work.**

The unit test on the TotScatCalculateSelfScattering workflow algorithm wasn't being run due to it being omitted from the CMakeLists.txt. This was pointed out by the ORNL team recently

I've enabled the unit test and I've also changed the test to run on a generic workspace created by WorkspaceCreationHelper rather than a POLARIS run. TotScatCalculateSelfScattering is already run on POLARIS run 98533 as part of the ISIS Powder system tests. There is value in having both the POLARIS system test and a separate unit test for the TotScatCalculateSelfScattering algorithm because the ISIS Powder reduction workflow does other processing in addition to TotScatCalculateSelfScattering eg it calls PDFFourierTransform.

This also required a small change to the WorkspaceCreationHelper unit to ensure the return type of the workspace was Workspace_sptr rather than Workspace2D_sptr

I've also made a change to the usage example in the documentation for the CalculatePlaczekSelfScattering v1 algorithm (which is called from TotScatCalculateSelfScattering) so that it runs successfully and runs the v1 algorithm. The v1 algorithm is the one that the POLARIS instrument scientists have validated.

I've currently added this to the release because it seems useful to have this extra test running and it doesn't have a functional impact on the users

**To test:**

Check the usage script on v1 of the CalculatePlaczekScattering algorithm runs successfully

Re the unit test - I suggest just reviewing the Jenkins console output on one of the main jobs to check the unit test suite TotScatCalculateSelfScatteringTest was run

*There is no associated issue.*

*This does not require release notes* because **it has no functional impact to the end user - activating test only**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
